auto-value-parcel/src/main/java/com/ryanharter/auto/value/parcel/AutoValueParcelExtension.java

```diff
@@ -82,9 +81,25 @@ public Boolean nullable() {
 
   @Override
   public boolean applicable(Context context) {
+    TypeMirror autoValueClass = context.autoValueClass().asType();
+    // Disallow manual implementation of the CREATOR instance
+    if (!needsCreator(context)) {
+      context.processingEnvironment().getMessager().printMessage(Diagnostic.Kind.ERROR,
```

Comment:
Sure. Works for me.

On Sun, Apr 17, 2016, 2:19 AM Bradley Campbell <notifications@github.com>
wrote:

> In
> auto-value-parcel/src/main/java/com/ryanharter/auto/value/parcel/AutoValueParcelExtension.java
> <https://github.com/rharter/auto-value-parcel/pull/50#discussion_r59978908>
> :
>
> > @@ -82,9 +81,25 @@ public Boolean nullable() {
> >
> >    @Override
> >    public boolean applicable(Context context) {
> > +    TypeMirror autoValueClass = context.autoValueClass().asType();
> > +    // Disallow manual implementation of the CREATOR instance
> > +    if (!needsCreator(context)) {
> > +      context.processingEnvironment().getMessager().printMessage(Diagnostic.Kind.ERROR,
>
> To get the variable element to pass to this exception, I'm thinking of
> changing the needsCreator method to return the variable element itself
> (or null). Your suggestion below was to change the name of needsCreator
> to hasCreator, but it might not make as much sense if it doesn't return a
> boolean anymore. Would findCreator work?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/rharter/auto-value-parcel/pull/50/files/24a0522bfe2135308d3075076c44125271223ad3#r59978908>
>